src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -285,12 +294,14 @@ CHIP_ERROR DefaultICDClientStorage::SetKey(ICDClientInfo & clientInfo, const Byt
     Crypto::Symmetric128BitsKeyByteArray keyMaterial;
     memcpy(keyMaterial, keyData.data(), sizeof(Crypto::Symmetric128BitsKeyByteArray));
 
-    return mpKeyStore->CreateKey(keyMaterial, clientInfo.shared_key);
+    ReturnErrorOnFailure(mpKeyStore->CreateKey(keyMaterial, clientInfo.aes_key_handle));
+    return mpKeyStore->CreateKey(keyMaterial, clientInfo.hmac_key_handle);
```

Comment:
If creating the hmac key handle fails, we need should delete the AES key imo to avoid having a dangling key until the next create key.

Reply:
make sense, updated, thanks